filebeat/beater/filebeat.go

```diff
@@ -106,6 +125,12 @@ func (fb *Filebeat) Run(b *beat.Beat) error {
 	// Blocks progressing. As soon as channel is closed, all defer statements come into play
 	<-fb.done
 
+	if fb.config.ShutdownTimeout > 0 {
+		// Wait for either timeout or all events having been ACKed by outputs.
```

Comment:
We should add here a log message that shutdown will wait for time X to shutdown or success